app/controllers/constraints_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class ConstraintsController < AuthenticationController
+  before_action :set_planning_application
+  before_action :ensure_constraint_edits_unlocked, only: %i[edit update]
+
+  def show; end
+
+  def edit; end
+
+  def update
+    @planning_application.constraints = params[:planning_application][:constraints].reject(&:blank?)
+    if @planning_application.save!
+      redirect_to @planning_application, notice: "Constraints have been updated"
+    else
+      render :edit
+    end
+  end
+
+  def check
+    @planning_application.constraints_checked!
```

Comment:
This will blow up if we don't rescue the error in the method unless that is the intention